napari/components/viewer_model.py

```diff
@@ -43,7 +45,9 @@ class ViewerModel(KeymapMixin):
 
     themes = palettes
 
-    def __init__(self, title='napari', ndisplay=2, order=None):
+    def __init__(
+        self, title='napari', ndisplay=2, order=None, axis_labels=None
+    ):
```

Comment:
I don't think this should hold this PR, it's more of a long term discussion, but I feel like axis labels should come with the data, not the viewer. Putting axis labels on the viewer before sending in the data requires users to keep track of axes at *two* places: viewer initialisation, and data input. And if your datasets have different axis ordering, then you really have to do a lot of gymnastics.

So, medium term, I think we should update this to be an attribute on the data/layer dims, rather than on the viewer. All this is of course making me think of xarray...

Reply:
yup, fair points, though will need more discussion and agreed can come in followup PRs

Reply:
I left it as is currently.